src/main/java/mezz/jei/ingredients/IngredientListElementComparator.java

```diff
@@ -1,41 +1,483 @@
 package mezz.jei.ingredients;
 
-import java.util.Comparator;
-
 import mezz.jei.config.Constants;
 import mezz.jei.gui.ingredients.IIngredientListElement;
 import net.minecraft.item.ItemStack;
+import net.minecraft.item.ItemTool;
+import mezz.jei.config.Config;
+import mezz.jei.util.Translator;
+import net.minecraft.entity.SharedMonsterAttributes;
+import net.minecraft.entity.ai.attributes.AttributeModifier;
+import net.minecraft.inventory.EntityEquipmentSlot;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemArmor;
+import net.minecraft.item.ItemHoe;
+import net.minecraft.item.ItemShears;
+import scala.Int;
```

Comment:
importing scala can cause issues

Reply:
I was only using it for Int.MaxValue().  Integer.MAX_VALUE should be used instead.